code/modules/atmospherics/machinery/atmosmachinery.dm

```diff
@@ -296,3 +294,7 @@ Pipelines + Other Objects -> Pipe network
 	if(node_turf.loc != self_turf.loc) //shuttles are area based, so this means the node is not on the shuttle with us
 		node.disconnect(src)
 		NODE_I = null
+
+/obj/machinery/atmospherics/update_remote_sight(mob/user)
+	user.sight |= (SEE_TURFS|BLIND)
+	user.update_pipe_vision(src)

```

Comment:
You no longer see mobs or objects while inside gas pipes, unless you have xray/thermal/etc...